miasm2/core/asmblock.py

```diff
@@ -606,45 +722,58 @@ def add_node(self, block):
                                           set()).add(to_add)
             else:
                 # Block is already in known nodes
-                self.add_edge(block, dst, constraint.c_t)
+                self.add_constraint(block.label, dst.label, constraint.c_t)
 
         return status
 
-    def del_node(self, block):
-        super(AsmCFG, self).del_node(block)
-        del self._label2block[block.label]
-
     def merge(self, graph):
         """Merge with @graph, taking in account constraints"""
         # -> add_edge(x, y, constraint)
-        for node in graph._nodes:
-            self.add_node(node)
+        self._loc_key_to_block.update(graph._loc_key_to_block)
```

Comment:
It is probably better to have the `add_block` handling `self._loc_key_to_block` update (and spot errors)